File: infrastructures/ddd/csharp/src/Api/Dto/V1/UpdateExampleRequest.cs
using System.ComponentModel.DataAnnotations;

namespace OddlyDdd.Api.Dto.V1
{
    /// <summary>
    /// Example request DTO for updating an example.
    /// </summary>
    public class UpdateExampleRequest : BaseRequestDto
    {
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name cannot exceed 100 characters")]
        public string Name { get; set; } = string.Empty;

        [StringLength(500, ErrorMessage = "Description cannot exceed 500 characters")]
        public string Description { get; set; } = string.Empty;
    }
}

File: infrastructures/ddd/python/src/api/dto/requests/update_example_request.py
"""Request DTO for updating an example."""

from dataclasses import dataclass
from api.dto.infra import BaseRequestDto


@dataclass
class UpdateExampleRequest(BaseRequestDto):
    """Request DTO for updating an existing example."""
    
    name: str = ""
    description: str = ""
    
    def validate(self) -> None:
        """
        Validates the request DTO.
        
        Raises:
            ValueError: If validation fails
        """
        if not self.name or not self.name.strip():
            raise ValueError("Name is required")
        
        if len(self.name) > 100:
            raise ValueError("Name cannot exceed 100 characters")
        
        if len(self.description) > 500:
            raise ValueError("Description cannot exceed 500 characters")
